lib/timers.js

```diff
@@ -264,7 +264,7 @@ exports.setInterval = function(callback, repeat) {
     timer._repeat.call(this);
 
     // Timer might be closed - no point in restarting it
-    if (!timer._repeat)
+    if ((!timer._repeat) || ((timer._idleTimeout === -1) && (!this._handle)))
```

Comment:
Nit: there's an extra set or parenthesis here.

Also I think the check should prioritize `timer._idleTimeout === -1`, and only check `._repeat` for legacy purposes.

I'm not entirely sure checking the handle is necessary. (Aside: `Timer#unref()` should probably check  `._idleTimeout === -1`, I *think*.)